OOP/OOP6_ResizableArray/src/main/java/LukCharList.java

```diff
@@ -0,0 +1,146 @@
+public class LukCharList implements CharList {
+
+    public char[] array = new char[0];
+    public char[] temporaryArray = new char[0];
+
+    public LukCharList(){
+    }
+
+    public int save(char charToSave) {
+        if(this.array.length > 0) {
+            temporaryArray = this.array;
+        }
+        this.array = new char[this.array.length + 1];
+        rewrite(temporaryArray);
+        array[this.array.length - 1] = charToSave;
+        return this.array.length - 1;
+    }
+
+    private char[] rewrite(char[] temporary) {
+        if(temporary.length > 0) {
+            for(int newLength = 0; newLength < temporary.length; newLength++) {
```

Comment:
what new length means in this context?

Reply:
@MatDls newLength because new array (length + 1). I know, newLength length() is the same like old but in this method we create rewrite array with +1 length.
This variable may be another name. For example 'i'

Reply:
@MatDls  I rename for lenghtOldArray because we rewrite old array